core/src/main/scala/akka/persistence/jdbc/journal/dao/ByteArrayJournalSerializer.scala

```diff
@@ -8,29 +8,38 @@ package journal.dao
 
 import akka.persistence.PersistentRepr
 import akka.persistence.jdbc.serialization.FlowPersistentReprSerializer
+import akka.persistence.jdbc.tag.EventTagConverter
 import akka.serialization.Serialization
 
 import scala.collection.immutable._
+import scala.concurrent.ExecutionContext
 import scala.util.Try
 
-class ByteArrayJournalSerializer(serialization: Serialization, separator: String)
+class ByteArrayJournalSerializer(serialization: Serialization, tagConverter: EventTagConverter)(implicit ctx: ExecutionContext)
     extends FlowPersistentReprSerializer[JournalRow] {
-  override def serialize(persistentRepr: PersistentRepr, tags: Set[String]): Try[JournalRow] = {
-    serialization
-      .serialize(persistentRepr)
-      .map(
-        JournalRow(
-          Long.MinValue,
-          persistentRepr.deleted,
-          persistentRepr.persistenceId,
-          persistentRepr.sequenceNr,
-          _,
-          encodeTags(tags, separator)))
-  }
-
-  override def deserialize(journalRow: JournalRow): Try[(PersistentRepr, Set[String], Long)] = {
-    serialization
-      .deserialize(journalRow.message, classOf[PersistentRepr])
-      .map((_, decodeTags(journalRow.tags, separator), journalRow.ordering))
-  }
+
+    override def serialize(persistentRepr: PersistentRepr, tags: Set[String]): Try[JournalRow] =
+      serialization
+        .serialize(persistentRepr)
+        .map(
+          JournalRow(
+            Long.MinValue,
+            persistentRepr.deleted,
+            persistentRepr.persistenceId,
+            persistentRepr.sequenceNr,
+            _,
+            tags.map(tagConverter.getIdByNameForce).toList
+          ))
+
+
+    override def deserialize(journalRow: JournalRow): Try[(PersistentRepr, Long)] =
+      serialization
+        .deserialize(journalRow.message, classOf[PersistentRepr])
+        .map((_, journalRow.ordering))
+
+}
+
+object ByteArrayJournalSerializer {
+  val tagToIdMap: Map[String, Int] = Map("firstEvent" -> 1, "longtag" -> 2, "multiT1" -> 3, "multiT2" -> 4, "firstUnique" -> 5, "tag" -> 6, "expected" -> 7, "multi" -> 8, "companion" -> 9, "xxx" -> 10, "ended" -> 11, "last" -> 12, "mul" -> 13)
+  val idToTagMap: Map[Int, String] = tagToIdMap.map { case (tag, id) => (id, tag) }
```

Comment:
Looks like some old code

Reply:
Indeed, this has to be removed. Good catch!